### ISSUE TITLE
Envelope with offline storage can't reupload

### ISSUE THREAD

**Issue by wangbax** [getsentry/sentry-java#2373]
### Integration

sentry-android

### Build System

Gradle

### AGP Version

7.0.2

### Proguard

Disabled

### Version

6.7.0

### Steps to Reproduce

1. close network
2. open app
3. close app
4. reopen app

### Expected Result

Adding all envelope to offline storage: `/data/user/0/[package]/cache/sentry/[dsn]/[id].envelope`

### Actual Result

Adding all envelope to offline storage: `/data/user/0/[package]/cache/sentry/[id].envelope`

<img width="900" alt="image" src="https://user-images.githubusercontent.com/5964735/202350469-aedd868b-3971-4e82-b75f-41ef4d5e950b.png">

**Comment by wangbax** [comment]
This issues actually is related to [new AndroidEnvelopeCache(options)](https://github.com/getsentry/sentry-java/blob/main/sentry-android-core/src/main/java/io/sentry/android/core/AndroidOptionsInitializer.java#L136) #2277 

<img width="902" alt="image" src="https://user-images.githubusercontent.com/5964735/202351021-be71e683-facc-4c41-9a6b-8566547e2e74.png">
<img width="1150" alt="image" src="https://user-images.githubusercontent.com/5964735/202351272-9133d429-ea0c-4804-9e46-11c065b982e2.png">

My solution: 
``` kotlin
SentryAndroid.init(context) { options ->
      options.apply {
        this.dsn = dsn
        // after change dsn, try reset disk cache
        setEnvelopeDiskCache(AndroidEnvelopeCache(options))
      }
    }
```

Hope can fix this issue least version. ☺️

**Comment by markushi** [comment]
FYI: The fix will be part of the `6.8.0` release.
